pkg/apis/submariner.io/v1/types.go

```diff
@@ -156,3 +156,97 @@ func (c *Connection) SetStatus(status ConnectionStatus, messageFormat string, a
 	c.Status = status
 	c.StatusMessage = fmt.Sprintf(messageFormat, a...)
 }
+
+// +genclient
+// +genclient:noStatus
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+type GlobalnetEgressIP struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// Spec is the specification of desired behavior of GlobalnetEgressIP object
+	Spec EgressIPSpec `json:"spec"`
+
+	// Observed status of GlobalnetEgressIP. Its a read-only field.
+	// +optional
+	Status EgressIPStatus `json:"status,omitempty"`
+}
+
+type EgressIPSpec struct {
+	// User can explicitly request a specific GlobalIP and this should be a valid IPaddress
+	// from globalnetCIDR allocated to the Cluster. If the requested globalIP falls outside
+	// of the globalnetCIDR or is already allocated (either fully or partially), then Status
+	// field will be set to Error with appropriate message.
+	// If this field is not specified, a single globalIP will be allocated.
+	// This field cannot be changed through updates.
+	// +optional
+	GlobalIP string `json:"globalIP,omitempty"`
+
+	// The number of globalIPs requested. Globalnet Controller will allocate the requested
+	// number of contiguous GlobalIPs for this GlobalnetEgressIP object.
+	// User can specify this field while omitting GlobalIP. In such cases, Globalnet will
+	// auto-allocate the requested number of contiguous GlobalIPs.
+	// If unspecified, NumGlobalIPs defaults to 1.
```

Comment:
I assume this is mutually exclusive with `GlobalIP` and that `GlobalIP` takes precedence. I also assume the IPs are allocated from the cluster's Globalnet CIDR.

```suggestion
	// The requested number of contiguous GlobalIPs to allocate from the Globalnet CIDR assigned to the cluster.
	// This field is ignored if the GlobalIP field is set. Defaults to 1.
```